src/modules/keyboardmanager/dll/dllmain.cpp

```diff
@@ -64,12 +66,79 @@ class KeyboardManager : public PowertoyModuleIface
+            auto current_config = settings.get_string_value(KeyboardManagerConstants::ActiveConfigurationSettingName);
+
+            if (current_config)
+            {
+                keyboardManagerState.SetCurrentConfigName(*current_config);
+                // Read the config file and load the remaps.
+                auto configFile = json::from_file(PTSettingsHelper::get_module_save_folder_location(KeyboardManagerConstants::ModuleName) + L"\\" + *current_config + L".json");
+                if (configFile)
+                {
+                    auto jsonData = *configFile;
+                    auto remapKeysData = jsonData.GetNamedArray(KeyboardManagerConstants::RemapKeysSettingName);
+                    auto remapShortcutsData = jsonData.GetNamedArray(KeyboardManagerConstants::RemapShortcutsSettingName);
+                    keyboardManagerState.ClearSingleKeyRemaps();
+
+                    if (remapKeysData)
+                    {
+                        for (const auto &it : remapKeysData)
+                        {
+                            try
+                            {
+                                auto originalKey = it.GetObjectW().GetNamedString(KeyboardManagerConstants::OriginalKeysSettingName);
+                                auto newRemapKey = it.GetObjectW().GetNamedString(KeyboardManagerConstants::NewRemapKeysSettingName);
+                                keyboardManagerState.AddSingleKeyRemap(std::stoul(originalKey.c_str()), std::stoul(newRemapKey.c_str()));
+                            }
+                            catch (...)
+                            {
+                                // Improper Key Data JSON. Try the next remap.
+                            }
+                        }
+                    }
+
+                    keyboardManagerState.ClearOSLevelShortcuts();
+                    if (remapShortcutsData)
+                    {
+                        for (const auto &it : remapShortcutsData)
+                        {
+                            try
+                            {
+                                auto originalKeys = it.GetObjectW().GetNamedString(KeyboardManagerConstants::OriginalKeysSettingName);
+                                auto newRemapKeys = it.GetObjectW().GetNamedString(KeyboardManagerConstants::NewRemapKeysSettingName);
+                                Shortcut originalSC(originalKeys.c_str());
+                                Shortcut newRemapSC(newRemapKeys.c_str());
+                                keyboardManagerState.AddOSLevelShortcut(originalSC, newRemapSC);
+                            }
+                            catch (...)
+                            {
+                                // Improper Key Data JSON. Try the next shortcut.
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        catch (...)
+        {
+            // Unable to load inital config.

```

Comment:
What is the current behaviour in this scenario? Like if we partially read the JSON file, and do some kbMState.AddOsLevelShortcut and then it gives error, will those be loaded in?

Reply:
Current logic is not to fail the whole load operation if there are some corrupted entries it will load partial entries. In case if there are some corrupted keys in the json we will delete those while saving the configuration.